File: SnapHack/context/samples/Spectacles Mobile Kit/sdk/iOS/SpectaclesKit/Sources/API/SessionException.swift
// Copyright © 2024 Snap, Inc. All rights reserved.

import Foundation

// MARK: - Session Exception Class

public class SessionException: Error, @unchecked Sendable {
    public let message: String
    public let cause: (any Error)?

    public init(message: String, cause: (any Error)? = nil) {
        self.message = message
        self.cause = cause
    }

    public class ConnectionFailed: SessionException {}
}

File: SnapHack/context/samples/Spectacles Mobile Kit/sdk/Android/SpectaclesKit/kit/src/main/java/com/snap/spectacles/kit/SessionException.kt
package com.snap.spectacles.kit

/**
 * Base exception class for SpectaclesSession-related errors.
 *
 * @param message The error message.
 * @param cause The underlying cause of the error, if any.
 */
open class SessionException(
    message: String,
    cause: Exception? = null
) : Exception(message, cause) {

    /**
     * Exception thrown when a connection attempt fails.
     */
    class ConnectionFailed(message: String) : SessionException(message)
}
